src/System.IO.FileSystem/tests/WinRT_BrokeredFunctions.cs

```diff
@@ -0,0 +1,207 @@
+            Assert.False(Directory.Exists(testFolder), "destination shouldn't exist");
+            Directory.CreateDirectory(testFolder);
+            Assert.True(Directory.Exists(testFolder), "destination should exist");
+            Directory.Delete(testFolder);
+            Assert.False(Directory.Exists(testFolder), "destination shouldn't exist");
+        }
+
+        [Fact]
+        public void ReplaceFile()
+        {
+            string testFile = GetTestFilePath();
+            File.Create(testFile).Dispose();
+
+            string destination = Path.Combine(s_musicFolder, "ReplaceFile_" + Path.GetRandomFileName());
+            File.Copy(testFile, destination);
+
+            // Need to be on the same drive
+            Assert.Equal(testFile[0], destination[0]);
+
+            string destinationBackup = Path.Combine(s_musicFolder, "ReplaceFile_" + Path.GetRandomFileName());
+            try
+            {
+                Assert.True(File.Exists(destination), "destination should exist before replacing");
+                Assert.False(File.Exists(destinationBackup), "destination shouldn't exist before replacing");
+                File.Replace(testFile, destination, destinationBackup);
+                Assert.False(File.Exists(testFile), "testFile shouldn't exist after replacing");
+                Assert.True(File.Exists(destination), "destination should exist after replacing");
+                Assert.True(File.Exists(destinationBackup), "destinationBackup should exist after replacing");
+            }
+            finally
+            {
+                File.Delete(destination);
+                File.Delete(destinationBackup);
+            }
+        }
+
+        [Fact]
+        public void SetFileAttributes()
+        {
+            string destination = Path.Combine(s_musicFolder, "SetFileAttributes_" + Path.GetRandomFileName());
+            CreateFileInBrokeredLocation(destination);
+            FileAttributes attr = File.GetAttributes(destination);
+            try
+            {
+                Assert.False(((attr & FileAttributes.ReadOnly) > 0), "new file in brokered location should not be readonly");
+                File.SetAttributes(destination, attr | FileAttributes.ReadOnly);
+                Assert.True(((File.GetAttributes(destination) & FileAttributes.ReadOnly) > 0), "file in brokered location should be readonly after setting FileAttributes");
+            }
+            finally
+            {
+                File.SetAttributes(destination, attr);
+                Assert.False(((File.GetAttributes(destination) & FileAttributes.ReadOnly) > 0), "file in brokered location should NOT be readonly after setting FileAttributes");
+                File.Delete(destination);
+            }
+        }
+
+        private void CreateFileInBrokeredLocation(string path)
+        {
+            // Temporary hack until FileStream is updated to support brokering
```

Comment:
No, as I won't check in until it works (e.g. CoreCLR is updated). I'll remove the comment/hack before merging.